ui/src/tempVars/components/TemplateControlDropdown.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {PureComponent} from 'react'
+
+import Dropdown from 'src/shared/components/Dropdown'
+import SimpleOverlay from 'src/shared/components/SimpleOverlay'
+import EditTemplateVariable from 'src/tempVars/components/EditTemplateVariable'
+import {calculateDropdownWidth} from 'src/dashboards/constants/templateControlBar'
+import Authorized, {isUserAuthorized, EDITOR_ROLE} from 'src/auth/Authorized'
+import {Template, Source} from 'src/types'
+
+interface Props {
+  template: Template
+  meRole: string
+  isUsingAuth: boolean
+  source: Source
+  onSelectTemplate: (id: string) => void
+  onCreateTemplate: (template: Template) => Promise<any>
+  onUpdateTemplate: (template: Template) => Promise<any>
+  onDeleteTemplate: (template: Template) => Promise<any>
+}
+
+interface State {
+  isEditing: boolean
+}
+
+export default class TemplateControlDropdown extends PureComponent<
+  Props,
+  State
+> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isEditing: false,
+    }
+  }
+
+  public render() {
+    const {
+      template,
+      isUsingAuth,
+      meRole,
+      source,
+      onSelectTemplate,
+      onCreateTemplate,
+    } = this.props
+    const {isEditing} = this.state
+
+    const dropdownItems = template.values.map(value => ({
+      ...value,
+      text: value.value,
+    }))
+
+    const dropdownStyle = template.values.length
+      ? {minWidth: calculateDropdownWidth(template.values)}
+      : null
+
+    const selectedItem = dropdownItems.find(item => item.selected) ||
```

Comment:
makes me wonder if the dropdown should handle this, in tandem with the new LoaderPlaceholder